pkg/controllermanager/apis/config/types.go

```diff
@@ -77,6 +77,44 @@ type CloudProfileControllerConfiguration struct {
 	// ConcurrentSyncs is the number of workers used for the controller to work on
 	// events.
 	ConcurrentSyncs int
+	// KubernetesVersionManagement configures the version policy that applies to
+	// Kubernetes versions in the CloudProfile
+	KubernetesVersionManagement *KubernetesVersionManagement
+	// MachineImageVersionManagement configures the version policy that applies to
+	// MachineImage versions in the CloudProfile
+	MachineImageVersionManagement *MachineImageVersionManagement
+}
+
+// KubernetesVersionManagement configures the version policy that applies to
+// Kubernetes versions in the CloudProfile
+// You can read more about it here: https://github.com/gardener/gardener/blob/master/docs/proposals/05-versioning-policy.md
+type KubernetesVersionManagement struct {
+	// Enabled defines whether the KubernetesVersionManagement is enabled
+	Enabled bool
+	// SupportedKubernetesVersions is the amount of minor Kubernetes versions that are considered to be "supported"
+	// refers to versions existing in the CloudProfile
+	// defaults to 3 as this is common practice in the Kubernetes Community
+	// e.g Versions in CloudProfile: 1.17, 1.15, 1.14, 1.13 -> Supported: 1.17 & 1.15 & 1.14, Unsupported: 1.13
+	SupportedKubernetesVersions *int
```

Comment:
The term `Supported` w.r.t. the Kubernetes major+minor version in contrast to the classifier `Supported` is kind of confusing and especially the code readability suffers from this. Do you think you can come up with another term?

Reply:
Renamed it to MaintainedKubernetesVersions to point out the difference to "SupportedVersions"